interp/src/debugger/commands.rs

```diff
@@ -0,0 +1,177 @@
+use std::fmt::Display;
+use std::fmt::Write;
+
+pub enum InterpreterError {
+    InvalidCommand(String), // this isn't used yet, but may be useful later when commands have more syntax
+    UnknownCommand(String),
+}
+
+impl Display for InterpreterError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let out_str = match self {
+            InterpreterError::InvalidCommand(msg) => {
+                format!("Invalid Command: {}", msg)
+            }
+            InterpreterError::UnknownCommand(s) => {
+                format!("Unknown command {}", s)
+            }
+        };
+        f.write_str(&out_str)
+    }
+}
+
+const HELP_LIST: [Command<&str>; 10] = [
+    Command::Step,
+    Command::Continue,
+    Command::Display,
+    Command::PrintOne(""),
+    Command::Break(""),
+    Command::Help,
+    Command::InfoBreak,
+    Command::DelByName(""),
+    Command::EnableByName(""),
+    Command::DisableByName(""),
+];
+pub enum Command<S: AsRef<str>> {
+    Step,                // Step execution
+    Continue,            // Execute until breakpoint
+    Empty,               // Empty command, does nothing
+    Display,             // Display full environment contents
+    PrintOne(S),         // Print a cell's ports
+    PrintTwo(S, S),      // Print a specific port or specific cell
+    PrintThree(S, S, S), // Print a specific port (fully specified)
```

Comment:
The names `PrintOne`, `PrintTwo`, ... aren't very descriptive. Change to reflect their semantics.

Reply:
I took a stab at it, see if you like the new names better. PrintTwo is a pain since it could be either a port or a cell depending on what is being specified